frontend/src/api/modelRegistry/errorUtils.ts

```diff
@@ -0,0 +1,12 @@
+import { ModelRegistryError } from '~/concepts/modelRegistry/types';
+
+const isError = (e: unknown): e is ModelRegistryError =>
+  ['code', 'message'].every((key) => key in (e as ModelRegistryError));
+
+export const handleModelRegistryFailures = <T>(promise: Promise<T>): Promise<T> =>
+  promise.then((result) => {
+    if (isError(result)) {
+      throw result;
+    }
+    return result;
+  });
```

Comment:
Is this a copy of the Pipeline code or does Model Registry really send back errors as 200?

Reply:
Model Registry doesn't send errors as 200. This is what I'm testing currently